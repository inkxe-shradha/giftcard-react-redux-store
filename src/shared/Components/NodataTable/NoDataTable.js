import React from "react";

const NoDataTable = ({ type = "received" }) => {
  return (
    <table className="table table-dark table-striped">
      <thead>
        <tr>
          <th>Card Name</th>
          <th>Points</th>
          <th>Card Desc</th>
          <th>{type === "received" ? "Received From" : "Sent To"}</th>
          <th>Issued Date</th>
          <th>Expiry Date</th>
          {type === "received" && <th>Actions</th>}
        </tr>
      </thead>
      <tbody>
        <tr>
          <td colSpan={type === "received" ? "7" : "6"} className="text-center">
            No Gift Card Received
          </td>
        </tr>
      </tbody>
    </table>
  );
};

export default NoDataTable;
