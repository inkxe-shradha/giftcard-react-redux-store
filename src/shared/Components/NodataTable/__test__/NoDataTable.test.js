import { render } from "@testing-library/react";
import { mount } from "enzyme";
import NoDataTable from "../NoDataTable";

describe("NoTable test suite case", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<NoDataTable />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should have 7 table head available when the props type passed as received", () => {
    const warper = mount(<NoDataTable type="received" />);
    expect(warper.find("th").length).toBe(7);
  });

  it("Should have 6 table head available when the props type passed as sent", () => {
    const warper = mount(<NoDataTable type="sent" />);
    expect(warper.find("th").length).toBe(6);
  });
});
