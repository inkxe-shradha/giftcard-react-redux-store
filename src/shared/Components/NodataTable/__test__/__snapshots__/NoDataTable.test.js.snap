// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoTable test suite case should render correctly 1`] = `
<DocumentFragment>
  <table
    class="table table-dark table-striped"
  >
    <thead>
      <tr>
        <th>
          Card Name
        </th>
        <th>
          Points
        </th>
        <th>
          Card Desc
        </th>
        <th>
          Received From
        </th>
        <th>
          Issued Date
        </th>
        <th>
          Expiry Date
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="text-center"
          colspan="7"
        >
          No Gift Card Received
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
